src/components/Species/DeleteSpeciesModal.tsx

```diff
@@ -1,16 +1,95 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Typography, useTheme } from '@mui/material';
 import strings from 'src/strings';
-import Button from '../common/button/Button';
-import DialogBox from '../common/DialogBox/DialogBox';
+import { BusySpinner, Button, DialogBox, TextTruncated } from '@terraware/web-components';
+import { SpeciesService } from 'src/services';
+import { SpeciesSearchResultRow } from './types';
+import { useOrganization } from 'src/providers';
 
 export interface DeleteSpeciesDialogProps {
   open: boolean;
   onClose: () => void;
-  onSubmit: () => void;
+  onSubmit: (toDelete: number[]) => void;
+  speciesToDelete: SpeciesSearchResultRow[];
 }
 
-export default function DeleteSpeciesDialog(props: DeleteSpeciesDialogProps): JSX.Element {
-  const { onClose, open, onSubmit } = props;
+export default function DeleteSpeciesDialog(props: DeleteSpeciesDialogProps): JSX.Element | null {
+  const { onClose, open, onSubmit, speciesToDelete } = props;
+  const theme = useTheme();
+  const { selectedOrganization } = useOrganization();
+  const [inUseSpecies, setInUseSpecies] = useState<Record<number, string>>();
```

Comment:
I thought the key type of a `Record` had to be a string (or a union type of string values) since JS objects always have string keys.

Reply:
The string/union type of strings is for utilities such as `Pick` or `Omit` where you pick keys from an existing type.
Record supports number in addition to strings, string literals. I suppose this is only for type checking, behind the scenes it is stored as a string.